File: coding/11th November 2025/Word Pattern.java
//Word Pattern
class Solution 
{
    public boolean wordPattern(String pattern, String s) 
    {
        char arr1[] = pattern.toCharArray();
        String arr2[] = s.split(" ");
        if(arr1.length != arr2.length)
            return false;
        HashMap<Character, String> map1 = new HashMap<Character, String>();
        HashMap<String, Character> map2 = new HashMap<String, Character>();
        for(int i = 0; i < arr1.length; i++)
        {
            if(map1.containsKey(arr1[i]) && !(map1.get(arr1[i]).equals(arr2[i])))
                return false;
            else
                map1.put(arr1[i], arr2[i]);

            if(map2.containsKey(arr2[i]) && !(map2.get(arr2[i]).equals(arr1[i])))
                return false;
            else
                map2.put(arr2[i], arr1[i]);
        }
        return true;
    }
}

File: Coding in Python/04th February 2026/Word Pattern.py
#Word Pattern
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        map1 = {}
        used = set()
        arr = s.split(' ')
        if(len(pattern) != len(arr)):
            return False
        for i in range(0, len(arr)):
            ch = pattern[i]
            if ch in map1:
                if map1.get(ch) != arr[i]:
                    return False
            else:
                if arr[i] in used:
                    return False
                used.add(arr[i])
                map1[ch] = arr[i]
        return True
